docs/cumulusci_flow.rst

```diff
@@ -0,0 +1,273 @@
+CumulusCI Flow
+==============
+CumulusCI Flow is the branching model and process we use for development of our products at Salesforce.org.
+CumulusCI Flow is based on the Github Flow model with a few changes and additions.
+
+CumulusCI Flow is implemented in the default flows provided by CumulusCI, but the approach to working with a Github repository does not require the use of CumulusCI.
+
+.. image:: images/salesforce-org-process.png
+   :scale: 50 %
+   :alt: Salesforce.org Dev/Release Process Diagram
+
+
+
+Project Considerations
+----------------------
+CumulusCI Flow was designed for use with Salesforce development projects, which inject some unique considerations into finding the right branching model:
+
+* You cannot re-cut a Salesforce managed package release with the same version as a prior release.  As a result, Git Tags are the best representation of our releases in a repository since they are a read only reference to the exact code we put into a given release.
+* Most Salesforce managed package projects use Push Upgrades to keep all users of the package on the latest release.  This eliminates the need to consider supporting multiple past versions.
+* Releasing managed packages has some overhead involved including manual checks by release managers to ensure nothing gets permanently locked into the package in a release.  As a result, true continuous delivery isn't an option.  Whether you're on a team that wants to deliver quickly (e.g. a two week sprint cycle) or at team that makes several larger releases a year, CumulusCI offers functionality to help cut releases for all products with any changes.
+
+
+
+Main Builds
+-----------
+The main goal of the CumulusCI Flow is to always have the main branch ready to cut into a package.
+This way, we can merge a fix and cut an emergency release at any time in the development process.
+
+To test that we can package main, we upload a beta release on every commit to main and then test that beta release in a variety of Salesforce org environments concurrently.
+This build ranges from 15 minutes to 2 hours depending on the project and a passing build is proof we can package main at any point in time.
+
+When the upload of the beta release is completed, the main branch is auto-merged_ into all open feature branches.
+New betas are published on Github as a Github Release along with automatically generated release notes created by parsing the body of all Pull Requests merged since the last production release
```

Comment:
```suggestion
New betas are published on GitHub as a GitHub Release, along with automatically generated release notes drawn from the content of the Pull Requests merged since the last production release.
```